### ISSUE TITLE
[Bug][Dialog] - Focus is not trapped within the component when non-focusable elements are inside

### ISSUE THREAD

**Issue by filipKovachev** [telerik/kendo-react#1993]
## I'm submitting a...

* Bug report 

## Current behavior
When there are no focusable elements inside a Dialog, the focus is not trapped within resulting in the possibility for the users to navigate items behind the overlay on the page.

## Expected behavior
The Dialog should trap the focus within itself so that no matter if there are focusable or non-focusable elements inside it the user cannot interact with other elements on the page. This is mentioned in the component spec.

## Minimal reproduction of the problem with instructions
1. Open this Stackblitz: https://stackblitz.com/edit/react-jf6rm9-cpvmtz?file=app%2Fmain.tsx
2. Navigate with Tab and notice how the focus goes to the inputs on the page.

 Reported in Ticket ID: 1640955

**Comment by kdikov82** [comment]
As a temporary workaround you can set a tabIndex to any of the elements within the Dialog:
https://stackblitz.com/edit/react-jf6rm9-bxfwuv?file=app%2Fmain.tsx

```
<Dialog
          title={'Please confirm'}
          onClose={toggleDialog}
          closeIcon={false}
          onKeyDown={handleKeyDown}
          autoFocus={true}
        >
          <p style={{ margin: '25px', textAlign: 'center' }} tabIndex={0}>
            Are you sure you want to continue?
          </p>
        </Dialog>
```

**Comment by zdravkov** [comment]
The issue is fixed in the dev npm version -https://stackblitz.com/edit/react-jf6rm9-7rsc8r - will be available in the next official version of KendoReact.
